script/src/type_id.rs

```diff
@@ -72,6 +72,14 @@ impl<'a> TypeIdSystemScript<'a> {
                 LittleEndian::write_u64(&mut buf, input.since);
                 blake2b.update(&buf[..]);
             }
+            // Hash output indices(actually there is only one index) for current
+            // script group as well, so we can generate cells with different types
+            // In one transaction
+            for index in &self.script_group.output_indices {
```

Comment:
Use only the first index is enough.

So the digest can be simplified into `[first_input.previous_output.tx_hash, first_input.previous_output.index, first_output.index]`

Reply:
Thanks the logic is updated now

Reply:
Actually some more thought: can we just use `transaction hash` + `index of first output in the group`? It seems like that's also up to the task but more simpler without needing serializing outpoints

Reply:
outpoint is required since outputs in the same transaction have the same previous transaction hash.

Reply:
Yes but we can use transaction hash + first index of output in the group. This way outputs ini the same transaction will have different index of output, hence different hash. Transaction hash only replaces outpoint of first previous input. The benefit here is that it decouples the hash from the actual outpoint structure.

Reply:
It seems like `transaction hash + index of first output in the group` is enough.

Reply:
Thinking again, transaction hash is not possible here. Since the type script included here will be included in transaction hash calculation, and to calculate type script argument, we will need transaction hash, another chicken-egg problem.